tests/unit/commands/samconfig/test_samconfig.py

```diff
@@ -0,0 +1,483 @@
+                ANY,
+                True,
+                "github.com",
+                "nodejs10.x",
+                "maven",
+                "myoutput",
+                "myname",
+                "apptemplate",
+                True,
+                '{"key": "value", "key2": "value2"}',
+            )
+
+    @patch("samcli.commands.validate.validate.do_cli")
+    def test_validate(self, do_cli_mock):
+        config_values = {"template_file": "mytemplate.yaml"}
+
+        with samconfig_parameters(["validate"], self.scratch_dir, **config_values) as config_path:
+
+            from samcli.commands.validate.validate import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(ANY, str(Path(os.getcwd(), "mytemplate.yaml")))
+
+    @patch("samcli.commands.build.command.do_cli")
+    def test_build(self, do_cli_mock):
+        config_values = {
+            "function_identifier": "foo",
+            "template_file": "mytemplate.yaml",
+            "base_dir": "basedir",
+            "build_dir": "builddir",
+            "use_container": True,
+            "manifest": "requirements.txt",
+            "docker_network": "mynetwork",
+            "skip_pull_image": True,
+            "parameter_overrides": "ParameterKey=Key,ParameterValue=Value ParameterKey=Key2,ParameterValue=Value2",
+        }
+
+        with samconfig_parameters(["build"], self.scratch_dir, **config_values) as config_path:
+
+            from samcli.commands.build.command import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(
+                "foo",
+                str(Path(os.getcwd(), "mytemplate.yaml")),
+                "basedir",
+                "builddir",
+                True,
+                True,
+                "requirements.txt",
+                "mynetwork",
+                True,
+                {"Key": "Value", "Key2": "Value2"},
+                None,
+            )
+
+    @patch("samcli.commands.local.invoke.cli.do_cli")
+    def test_local_invoke(self, do_cli_mock):
+        config_values = {
+            "function_identifier": "foo",
+            "template_file": "mytemplate.yaml",
+            "event": "event",
+            "no_event": False,
+            "env_vars": "envvar.json",
+            "debug_port": [1, 2, 3],
```

Comment:
+1 glad to know multiples work well.

Reply:
lol, I know! Do we document this case explicitly?